Marlin/src/lcd/ultralcd.cpp

```diff
@@ -3678,7 +3682,10 @@ void kill_screen(const char* lcd_msg) {
   void lcd_control_filament_menu() {
     START_MENU();
     MENU_BACK(MSG_CONTROL);
-
+    #if ENABLED(ADVANCED_PAUSE_SPOOL_SWAP)//steeve
+	  MENU_ITEM_EDIT(bool,MSG_SWAP_SPOOL_ON, &swap_spool_enabled);
+	  MENU_ITEM_EDIT(int3,MSG_SWAP_SPOOL_STOP, &swap_spool_stop,active_extruder,FIL_RUNOUT_SENSORS);
+    #endif																				 
     #if ENABLED(LIN_ADVANCE)
       MENU_ITEM_EDIT(float3, MSG_ADVANCE_K, &planner.extruder_advance_k, 0, 999);
     #endif
```

Comment:
THANK YOU VERY MUCH

For everybody have helped me  'Tmc / Roxanne / tanoo / and thinkyhead my favorite teacher that becomes red when sees spaces or amazing poor indentations lollllllllllllllllllllll

Reply:
Funny you should mention that...   That indentation is wrong....
```cpp
+    #if ENABLED(ADVANCED_PAUSE_SPOOL_SWAP)//steeve
+	  MENU_ITEM_EDIT(bool,MSG_SWAP_SPOOL_ON, &swap_spool_enabled);
+	  MENU_ITEM_EDIT(int3,MSG_SWAP_SPOOL_STOP, &swap_spool_stop,active_extruder,FIL_RUNOUT_SENSORS);
+    #endif	
```

Reply:
Yes , but it's my signature , when i make a mouse copy , i loose indentations sometimes
I know thinkyhead will never merge this feature because , it needs to have the ' thinkyhead writting style ' ,but the code WORKS PERFECT
and It's enough for him 

Lolll

Reply:
Roxanne , did you have the time to test the issue   ' ubl/fwretract ' , i'm sure , it was only the fw and i'm sure now with the new merged code , all is solved 
Give me the good news about it

Reply:
@Roxy-3D 
The multi sensors , is primitive 
One sensors , for each spool/extruder
Or if not defined , one pin for all sensors wired together
For mixing , there is no reason to check sensors separatly , all wired together and one pin

The multi sensors have utility only for ' spool swapping' to give the runout protection for each new used extruder/spool , without checking the old always ON

Reply:
@thinkyhead 
I would change ' G27 ' , to add ' if xpos or ypos not defined use the current ' , but , G27 code is a jungle , too much classes for make less things than a M600 that is really compact
So i ask you to find the solution ,

Reply:
THE END !

Reply:
> I know thinkyhead will never merge this feature because , it needs to have the ' thinkyhead writting style ' 

I didn't write the [Coding Guidelines](http://marlinfw.org/docs/development/coding_standards.html) entirely for my entertainment. I wrote them for you so that I don't have to clean up vomit. Don't scoff at them.

Reply:
Hey man 😄  , relax ! 😄 
If I was engineer , with high experience in this world , and I tell you this way , it's sure , it would not a pleasure , but I'm a beginner , and you are a boss for me and all I say is only 'humor ', everytime .
So , when i say something about you , you can laugh every time because it is said for laughing
'The thinkyHead style' is required , no doubt ! But it's a bigger job to find all forgotten spaces than to code something lolll
And for me , the ' ThinkyHead Style ' is not the ' gangnam style ' , it's a true guideline for me , and i give this name because , you are the big man that is my boss , that corrects my errors , that make a beautifull firmware and manage it perfectly

I will give you a crow , if you implement a web server on ethernet boards !

Reply:
It's the only things that is missing on marlin
The web server !
And multi sensors for mixing machine , but , not a real problem lollllll

Reply:
**GREEN LIGHT !!!!!!!!!!!!!!!!!!!!**

There is the ' THINKYHEAD WRITING STYLE '
But
There is ' TRAVIS WRITING STYLE' too
Informatic is really not easy .. 😄